### PR TITLE
An idempotent property setter

### PR DESCRIPTION
**Pull request by rlaveycal**
### Description

The `iis_config` resource is not idempotent - it just runs `appcmd`. This resource uses the Powershell Get/Set cmdlets to provide idempotency.

### CODE REVIEW

**Comment by EasyAsABC123** [comment]
@rlaveycal nice work! i'll finish reviewing this after work today.

**Comment by EasyAsABC123** [comment]
@tas50 want to override the failure on appveyor?

File: libraries/matcher.rb [hunk 1]
```diff
@@ -7,6 +7,13 @@
                    )
   end
 
+  [:set, :add, :remove].each do |action|
```
Comment (tas50) [inline libraries/matcher.rb]: Probably time to kill off the matchers file. Modern ChefSpec in ChefDK generates these on the fly.

File: resources/config_property.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Configures an IIS property (using powershell for idempotence)
+
+property :property, String, name_property: true, required: true
```
Comment (tas50) [inline resources/config_property.rb]: requiring a name property doesn't make a whole lot of sense in a resource. The whole point of the name property is you would use the resource name *or* you could override it by setting this property. They should always be optional.

**Comment by rlaveycal** [comment]
@EasyAsABC123 Are you waiting for anything from me?

**Comment by EasyAsABC123** [comment]
@rlaveycal nope, @tas50 can we override the build failure since it appears to be a non-error?

**Comment by tas50** [comment]
Sorry for completely dropping this on the floor. The merge is done now.

**Pull request merged**